py/desitarget/cuts.py

```diff
@@ -986,89 +986,71 @@ def isBGS_colors(gflux=None, rflux=None, zflux=None, w1flux=None, w2flux=None,
     return bgs
 
 
-def isQSO_colors(gflux, rflux, zflux, w1flux, w2flux, optical=False, south=True):
-    """Convenience function for backwards-compatability prior to north/south split.
+def isQSO_cuts(gflux=None, rflux=None, zflux=None, w1flux=None, w2flux=None, 
+               w1snr=None, w2snr=None, deltaChi2=None, brightstarinblob=None, 
+               release=None, objtype=None, primary=None, optical=False, south=True):
+    """Definition of QSO target classes from color cuts. Returns a boolean array.
 
-    Args:
-        gflux, rflux, zflux, w1flux, w2flux: array_like
-            The flux in nano-maggies of g, r, z, W1, and W2 bands.
-        optical: boolean, defaults to False
-            Just apply optical color-cuts
-        south: boolean, defaults to ``True``
-            Call isQSO_colors_north if ``south=False``, otherwise call isQSO_colors_south.
+    Parameters
+    ----------
+    optical : :class:`boolean`, defaults to ``False``
+        Apply just optical color-cuts.
+    south : :class:`boolean`, defaults to ``True``
+        Use cuts appropriate to the Northern imaging surveys (BASS/MzLS) if ``south=False``,
+        otherwise use cuts appropriate to the Southern imaging survey (DECaLS).
 
-    Returns:
-        mask : array_like. True if the object has QSO-like colors.
+    Returns
+    -------
+    :class:`array_like`
+        ``True`` for objects that pass the quasar color/morphology/logic cuts.
+
+    Notes
+    -----
+    - Current version (12/07/18) is version 159 on `the wiki`_.
+    - See :func:`~desitarget.cuts.set_target_bits` for other parameters.
     """
-    if south is False:
-        return isQSO_colors_north(gflux, rflux, zflux, w1flux, w2flux,
-                                  optical=optical)
-    else:
-        return isQSO_colors_south(gflux, rflux, zflux, w1flux, w2flux,
-                                  optical=optical)
 
+    if not south:
+        gflux, rflux, zflux = shift_photo_north(gflux, rflux, zflux)
 
```

Comment:
I thought we had trained distinct color-cuts in the north vs the south rather than transforming the colors like this. I'm sure I'm just misremembering.

Also, I have a pass-by-reference vs pass-by-value question: since you're using the same variable names here won't the *input* grz fluxes be modified after this script is called and after they've been shifted to a different photometric system?  Or do the changes remain within the function (I honestly can't remember what happens in this case in python).

Reply:
The collaboration as a whole certainly expressed a preference for training color cuts rather than transforming the colors, and we've removed transformations from all of the other cuts except for the quasars. This shift was added by @CAClaveau [here](https://github.com/desihub/desitarget/commit/3345acd9ea799f0f7b6c384b98b6df00ecf52e8a), presumably as a placeholder. Although, we could ask @CAClaveau and @Cyeche to ensure that this shift is reverted to distinct cuts in the future.

The pass-by-reference vs pass-by-value question is a good one. I'm pretty sure that the changes remain within the function...but just to check I made a version of the code that copied `rflux`, `gflux` and `zflux` before calling `isQSO_cuts` and then compared the output `rflux`, `gflux` and `zflux` to the copies. They were identical.